itests/fevm_test.go

```diff
@@ -1436,3 +1436,97 @@ func TestEthGetBlockByNumber(t *testing.T) {
 	require.NotNil(t, pendingBlock)
 	require.True(t, pendingBlock.Number >= latest)
 }
+
+func TestEthGetTransactionByBlockHashAndIndexAndNumber(t *testing.T) {
+	ctx, cancel, client := kit.SetupFEVMTest(t)
+	defer cancel()
+
+	key, ethAddr, filAddr := client.EVM().NewAccount()
+	kit.SendFunds(ctx, t, client, filAddr, types.FromFil(10))
+
+	// Deploy multiple contracts in the same tipset
+	var txHashes []ethtypes.EthHash
+	var receipts []*api.EthTxReceipt
+	numTx := 3
+
+	for i := 0; i < numTx; i++ {
+		tx := deployContractWithEth(ctx, t, client, ethAddr, "./contracts/MultipleEvents.hex")
+		tx.Nonce = i
+		client.EVM().SignTransaction(tx, key.PrivateKey)
+		hash := client.EVM().SubmitTransaction(ctx, tx)
+		txHashes = append(txHashes, hash)
+	}
+
+	// Wait for all transactions to be mined
+	for _, hash := range txHashes {
+		receipt, err := client.EVM().WaitTransaction(ctx, hash)
+		require.NoError(t, err)
+		require.NotNil(t, receipt)
+		require.Equal(t, ethtypes.EthUint64(1), receipt.Status)
+		receipts = append(receipts, receipt)
+	}
+
+	// Ensure all transactions are in the same tipset
+	blockHash := receipts[0].BlockHash
+	blockNumber := receipts[0].BlockNumber
+	for _, receipt := range receipts[1:] {
+		require.Equal(t, blockHash, receipt.BlockHash)
+		require.Equal(t, blockNumber, receipt.BlockNumber)
+	}
```

Comment:
this is going to be flaky, because they legitimately may not be, you're depending on `SubmitTransaction` to queue up and all be accepted in the same tipset which you don't have a guarantee of.

You'll need to put this, and the 3 loops before it, into a loop, and repeat until they are all in the same tipset - you could just do that in the `WaitTransaction` loop - get the first one, then check its block against the subsequent ones. If they don't match, `continue`, otherwise `break`.